src/ray/stats/metric_exporter_client.cc

```diff
@@ -80,5 +81,99 @@ void MetricsAgentExporter::ReportMetrics(const std::vector<MetricPoint> &points)
       });
 }
 
+void MetricsAgentExporter::ReportMetrics(
+    const std::vector<std::pair<opencensus::stats::ViewDescriptor,
+                                opencensus::stats::ViewData>> &data) {
+  rpc::ReportOCMetricsRequest request_proto;
+  for (auto datum : data) {
+    auto view_descriptor = datum.first;
+    auto view_data = datum.second;
+    auto measure_descriptor = view_descriptor.measure_descriptor();
+
+    auto request_point_proto = request_proto.add_metrics();
+
+    auto metric_descriptor_proto = request_point_proto->mutable_metric_descriptor();
+    metric_descriptor_proto->set_name(measure_descriptor.name());
+    metric_descriptor_proto->set_description(measure_descriptor.description());
+    metric_descriptor_proto->set_unit(measure_descriptor.units());
+    for (auto &tag_key : view_descriptor.columns()) {
+      metric_descriptor_proto->add_label_keys()->set_key(tag_key.name());
+    };
+
+    auto start_time = absl::ToUnixSeconds(view_data.start_time());
+    auto end_time = absl::ToUnixSeconds(view_data.end_time());
+
+    auto make_new_data_point_proto = [request_point_proto, start_time,
+                                      end_time](std::vector<std::string> tag_values) {
+      auto metric_timeseries_proto = request_point_proto->add_timeseries();
+      metric_timeseries_proto->mutable_start_timestamp()->set_seconds(start_time);
+
+      for (auto value : tag_values) {
+        metric_timeseries_proto->add_label_values()->set_value(value);
+      };
+
+      auto point_proto = metric_timeseries_proto->add_points();
+      point_proto->mutable_timestamp()->set_seconds(end_time);
+      return point_proto;
+    };
+
+    switch (view_data.type()) {
+    case opencensus::stats::ViewData::Type::kDouble:
+      for (auto &row : view_data.double_data()) {
+        std::vector<std::string> tag_values = row.first;
+        double point_value = row.second;
+
+        auto point_proto = make_new_data_point_proto(tag_values);
+        point_proto->set_double_value(point_value);
+      }
+      break;
+    case opencensus::stats::ViewData::Type::kInt64:
+      for (auto &row : view_data.int_data()) {
+        auto tag_values = row.first;
+        int64_t point_value = row.second;
+
+        auto point_proto = make_new_data_point_proto(tag_values);
+        point_proto->set_int64_value(point_value);
+      }
+      break;
+    case opencensus::stats::ViewData::Type::kDistribution:
+      for (auto &row : view_data.distribution_data()) {
+        auto tag_values = row.first;
+        opencensus::stats::Distribution dist_value = row.second;
+
+        auto point_proto = make_new_data_point_proto(tag_values);
+        point_proto->mutable_timestamp()->set_seconds(end_time);
+        auto distribution_proto = point_proto->mutable_distribution_value();
+        distribution_proto->set_count(dist_value.count());
+        distribution_proto->set_sum(dist_value.count() * dist_value.mean());
```

Comment:
I found this cout/sum and other statistics metrics are invalid because it's only one opoint in view. You may have a try and see what I said.

Reply:
count and sum can still be aggregated. For latencies, it does help us to break down average latency per service and the sum of squared deviation tells me at the point in time how variable is the latency.

Reply:
Count and sum also look good for me. I mean there are no such complicated aggregation functions like P99/P95. If we cannot get all statistic measurements, why not calculating them in aggragator agent, which is reason I just want to collect raw data from opencesus. It's hard to get all what we want from reporter if no code change. @rkooo567